src/modules/SdnDiag.Utilities/private/Wait-PSJob.ps1

```diff
@@ -46,6 +46,8 @@ function Wait-PSJob {
 
             # check the stopwatch and break out of loop if we hit execution timeout limit
             if ($stopWatch.Elapsed.TotalSeconds -ge $ExecutionTimeOut) {
+                $stopWatch.Stop()
```

Comment:
Just curious why do we need this ? When we throw this would be disposed anyways , unless Im missing something ?

Reply:
So I was going off GitHub CoPilot and it mentions that Stopwatch does not have implement IDisposable. It recommends regardless to call .Stop() to not leave any lingering timers.

Since this section will throw a terminating error, it will not call the .Stop() method that is outside the while loop. Just figure since I noticed, easy enough to add the code to ensure it gets stopped.